packages/lit-dev-content/samples/examples/async-task/npm-info.ts

```diff
@@ -0,0 +1,109 @@
+import {LitElement, css, html} from 'lit';
+import {customElement, state} from 'lit/decorators.js';
+import {Task} from '@lit-labs/task';
+
+@customElement('npm-info')
+export class NpmInfo extends LitElement {
+  @state()
+  _package = 'lit';
+
+  _fetchNpmInfoTask = new Task(this, {
+    task: async ([pkgName], {signal}) => {
+      if (pkgName === undefined || pkgName === '') {
+        throw new Error('Empty package name');
+      }
+      // Artifical delay for demo purposes
+      await new Promise((r) => setTimeout(r, 1000));
+      const response = await fetch(getPackageUrl(pkgName), {signal});
+      if (response.status === 200) {
+        return response.json() as Promise<NpmPackage>;
+      } else {
+        throw response.text();
+      }
+    },
+    args: () => [this._package],
+  });
+
+  _onChange(e: Event) {
+    this._package = (e.target as HTMLInputElement).value;
+  }
+
+  render() {
+    return html`
+      <header>
+        <h1>${this._package}</h1>
+        <a>${npmLogo}</a>
+      </header>
+      <label>
+        Enter a package name:
+        <input .value=${this._package} @change=${this._onChange} />
+      </label>
+      <div>
+        ${this._fetchNpmInfoTask.render({
+          pending: () => 'Loading...',
+          complete: (pkg) => html`
+            <h3>${pkg.description}</h3>
+            <h4>dist-tags:</h4>
+            <ul>
+              ${Array.from(Object.entries(pkg['dist-tags'])).map(
+                ([tag, version]) => html`<li><pre>${tag}: ${version}</pre></li>`
+              )}
+            </ul>
+          `,
+          error: (e) => `Error ${(e as Error).message}`,
+        })}
+      </div>
+    `;
+  }
+
+  static styles = css`
+    :host {
+      display: block;
+      min-width: 300px;
+      border-radius: 5px;
+      border: solid 1px #aaa;
+      padding: 20px;
+    }
+    header {
+      display: flex;
+      flex-direction: row;
+      align-items: center;
+      justify-content: space-between;
+    }
+    #logo {
+      height: 38px;
+      width: auto;
+    }
+  `;
+}
+
+const getPackageUrl = (name: string) => `https://registry.npmjs.org/${name}`;
+
+const npmLogo = html`
```

Comment:
Could we find a live URL for this instead?

Reply:
done